public/src/components/Home/SidebarComponent.jsx

```diff
@@ -0,0 +1,144 @@
+import React from 'react';
+import ArchivedSubredditsComponent from './ArchivedSubredditsComponent';
+
+const ScrollArea = require('react-scrollbar').default;
+
+export default class SidebarComponent extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            searchFormValue: '',
+            archiveFormValue: '',
+        };
+        this.getSubredditsList = this.getSubredditsList.bind(this);
+        this.onChangeSearchForm = this.onChangeSearchForm.bind(this);
+        this.onChangeArchiveForm = this.onChangeArchiveForm.bind(this);
+        this.getArchiveFormClass = this.getArchiveFormClass.bind(this);
+        this.onKeyPressSearchForm = this.onKeyPressSearchForm.bind(this);
+        this.setSearchValue = this.setSearchValue.bind(this);
+    }
+
+    onKeyPressSearchForm(event) {
+        const { searchFormValue } = this.state;
+        const { storeSubredditToArchive } = this.props;
+        if (searchFormValue && event.key === 'Enter') {
+            storeSubredditToArchive(searchFormValue);
+            this.setState({
+                searchFormValue: '',
+            });
+        }
+    }
+
+    onChangeSearchForm(event) {
+        this.setState({
+            searchFormValue: event.target.value,
+        });
+    }
+
+    onChangeArchiveForm(event) {
+        this.setState({
+            archiveFormValue: event.target.value,
+        });
+    }
+
+    setSearchValue(value) {
+        this.setState({
+            searchFormValue: value,
+        });
+    }
+
+    getArchiveFormClass() {
+        const { subreddits } = this.props;
+        const minimumArchivedLength = 5;
```

Comment:
Rename to `MINIMUM_ARCHIVED_LENGTH` and move it after imports?

Reply:
fixed